gettsim/benefits/wohngeld_dag.py

```diff
@@ -270,3 +274,42 @@ def _wohngeld_eink_abzüge_ab_2016(
         abzüge = pd.Series(dtype=float)
 
     return abzüge
+
+
+def _wohngeld_eink(
+    tu_id,
+    haushalts_größe,
+    _wohngeld_eink_abzüge,
+    _wohngeld_abzüge,
+    _wohngeld_brutto_eink,
+    _wohngeld_sonstiges_eink,
+    wohngeld_params,
+):
+    _wohngeld_eink_abzüge_per_tu = _wohngeld_eink_abzüge.groupby(tu_id).transform("sum")
+
+    vorläufiges_eink = (1 - _wohngeld_abzüge) * (
+        _wohngeld_brutto_eink + _wohngeld_sonstiges_eink - _wohngeld_eink_abzüge_per_tu
+    )
+
+    unteres_eink = [
+        wohngeld_params["min_eink"][hh_size]
+        if hh_size < 12
+        else wohngeld_params["min_eink"][12]
+        for hh_size in haushalts_größe
+    ]
+
+    return vorläufiges_eink.clip(lower=unteres_eink)
+
+
+def haushalts_größe(hh_id):
+    return hh_id.groupby(hh_id).transform("size")
```

Comment:
What would this be? An index or a column? Or a column with the same index? I do not understand the construct of grouping by the same object, I guess.

Reply:
I am not sure what you mean exactly. I would say it is a column. Grouping by `hh_id` puts all the same `hh_id`s in the same group. Size yields the size of the group and `transform` converts the aggregated one-result-per-group back to the original index of `hh_id`, `hh_id.index`, assigning the size to each observation in the household.

Reply:
Okay, I guess I just have never used it on a series before, just on df's where you always pass a string column/index name. With this in mind it just looked weird to me grouping by an object itself.